keras/src/models/model_test.py

```diff
@@ -171,6 +173,18 @@ def test_functional_single_output(self, model_fn, loss_type):
         )
         self.assertListEqual(hist_keys, ref_keys)
 
+        reloaded_pickle = pickle.loads(pickle.dumps(model))
```

Comment:
Please add a separate unit test rather than inlining pickle testing here

Reply:
good point - added a unique test to cover this